File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution(object):
    def longestPalindrome(self, s):
        start = 0
        strLen = len(s)
        end = start + strLen
        notPal = True

        while notPal:
            if s == s[::-1]:
                notPal = False
                return s
                break
            while end != len(s) and notPal == True:
                string = s[start:end+1]
                if string == string[::-1]:
                    notPal = False
                    return string
                start += 1
                end = start + strLen
            strLen -= 1
            start = 0
            end = start + strLen

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string expand(string& s, int i, int j) {
        while (i >= 0 && j < s.size() && s[i] == s[j]) {
            i--; j++;
        }
        return s.substr(i + 1, j - i - 1);
    }

    string longestPalindrome(string s) {
        string ans = "";

        for (int i = 0; i < s.size(); i++) {
            string even = expand(s, i, i + 1);
            if (even.size() > ans.size())
                ans = even;

            string odd = expand(s, i, i);
            if (odd.size() > ans.size())
                ans = odd;
        }

        return ans;
    }
};
